### PR TITLE
Complete refactoring of the project

### PR DESCRIPTION
**Pull request by davidbgk**
Oops, I did it again 🙈 

Mostly moving parts back and forth but also:

* switch to blueprints everywhere
* cleaner declaration/use of `app`
* bases to iterate on tests' performances
* `assert 'html_fragment' in response` now works seamlessly in tests 🍾

In progress PR will be a nightmare to merge but I can hopefully handle it…

### CODE REVIEW

File: docs/architecture/decisions/0005-flask-architecture.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Plus, to deal with configurations within extensions is harder if you do not
+declare the Flask app once and for all then load the configuration.
+
+## Decision
+
+An architecture based on https://github.com/stephane/flask-skeleton
+
+## Consequences
+
+Beaking change, mostly for configurations.
```
Comment (vinyll) [inline docs/architecture/decisions/0005-flask-architecture.md:23]: [Beaking](https://www.urbandictionary.com/define.php?term=beaking) or _Breaking_

File: cafebabel/core/testing.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from flask import Response
+
+
+class ContainsResponse(Response):
```
Comment (vinyll) [inline cafebabel/core/testing.py:4]: 👌

File: cafebabel/__init__.py [hunk 3]
```diff
@@ -1,44 +1,111 @@
...
+    if app.debug:
+        try:
+            from flask_debugtoolbar import DebugToolbarExtension
+            DebugToolbarExtension(app)
+        except ImportError:
+            pass
+
+    if app.testing:
+        from cafebabel.core.testing import ContainsResponse
+        app.response_class = ContainsResponse
```
Comment (vinyll) [inline cafebabel/__init__.py:45]: 💣

File: cafebabel/tests/test_drafts.py [hunk 4]
```diff
@@ -66,15 +63,15 @@ def test_draft_editing_should_update_content(client, editor):
...
+def test_draft_image_should_save_and_render(app, client, editor):
     login(client, editor.email, 'secret')
     with open(Path(__file__).parent / 'dummy-image.jpg', 'rb') as content:
         image = BytesIO(content.read())
     data = {
         'title': 'My article',
         'summary': 'Summary',
         'body': 'Article body',
-        'language': 'en',
+        'language': 'pr',
```
Comment (vinyll) [inline cafebabel/tests/test_drafts.py]: Should this pass with a language that does not exist?

File: instance/config.local.py [hunk 5]
```diff
@@ -0,0 +1 @@
+FOO = 'bar'
```
Comment (vinyll) [inline instance/config.local.py:1]: Not sure why this is here?

File: docs/architecture/decisions/0005-flask-architecture.md (hunk 1 above)
Comment (davidbgk) [reply docs/architecture/decisions/0005-flask-architecture.md:23]: Typo, but I almost want to let it :D

Fixed in https://github.com/cafebabel/cafebabel.com/commit/9afda5e8bd76840a2fdbbf9f0a1a9f8f0956224b

File: instance/config.local.py (hunk 5 above)
Comment (davidbgk) [reply instance/config.local.py:1]: Just as an example of how to set up custom settings

**Pull request merged**